### PR TITLE
add support for actions to protected view

### PR DESCRIPTION
**Pull request by ori-shalom**

### CODE REVIEW

File: sdk/forms/README.md [hunk 1]
```diff
@@ -176,11 +176,11 @@ const form = createProtectedForm('#form-container', {
 
 #### Overview
 
-`createProtectedView` creates a secure view component to display sensitive data stored in Piiano Vault, ensuring that sensitive data is never directly exposed to the frontend.
+`createProtectedView` creates a secure view component to display sensitive data returned from Piiano Vault, ensuring that sensitive data is never directly exposed to the frontend.
```
Comment (motymichaely) [inline sdk/forms/README.md:179]: This sentence is confusing.  How would displaying sensitive data in the frontend ensures that sensitive data is never directly exposed to the frontend?

Comment (motymichaely) [reply sdk/forms/README.md:179]: It is also redundant with the "Goal".

File: sdk/forms/README.md [hunk 2]
```diff
@@ -190,14 +190,23 @@ Use `createProtectedView` when you need to securely display sensitive data on th
...
-    The order of properties in the array determines the order in which they will be displayed in the view.
-  - `transformationParam` (optional string): An extra transformation param to be sent to the Vault and be available in the transformation functions. When multiple parameters are needed, they can be passed as a JSON string and parsed in the transformation functions.
+  - `strategy` (object): Strategy options for getting data from the Vault:
+    - `type` (string): Type of the strategy. Can be either `"read-objects"` or `"invoke-action"`.
+      The rest of the configuration depends on the strategy type:
+      - When type is `"read-objects"` the view will read objects from the Vault and display them in the view. The following additional strategy options are required:
+        - `collection` (string): Name of the collection containing the data.
+        - `ids` (string array): IDs of objects to be displayed.
+        - `props` (string array): Properties to display from each object.
+          Each property can defined as is or with a transformation (e.g., `name`, `email`, `email.mask`, etc.).
```
Comment (motymichaely) [inline sdk/forms/README.md:200]:
```suggestion
          Each property can be provided as is or with a transformation (e.g., `name`, `email`, `email.mask`, etc.).
```

File: sdk/forms/README.md [hunk 3]
```diff
@@ -190,14 +190,23 @@ Use `createProtectedView` when you need to securely display sensitive data on th
...
+          The order of properties in the array determines the order in which they will be displayed in the view.
+        - `transformationParam` (optional string): An extra transformation param to be sent to the Vault and be available in the transformation functions. When multiple parameters are needed, they can be passed as a JSON string and parsed in the transformation functions.
+        - `reason` (string): Reason for accessing the data (will be logged in the Vault audit logs).
+      - When type is `"invoke-action"` the view will invoke an action in the Vault and display the result in the view. The following additional strategy options are required:
+        - `action` (string): Name of the action to invoke.
+        - `globalIdentifierParameters` (Record<string, string>): Vault global identifiers to be evaluated by the vault and passed to the action.
+        - `extraParameters` (Record<string, unknown>): An extra parameters to be sent to the action and be available in the action. These parameters are not evaluated by the vault before passing them to the action.
```
Comment (motymichaely) [inline sdk/forms/README.md:207]:
```suggestion
        - `extraParameters` (Record<string, unknown>): Extra parameters to be sent to the action and be available in the action. These parameters are not evaluated by the vault before passing them to the action.
```

**Review by motymichaely — DISMISSED** [review]

File: sdk/vault-bundles/src/globals.ts [hunk 4]
```diff
@@ -1,8 +1,39 @@
...
+   * API calls made with this function are subject to the same restrictions as other outbound network requests in Vault and must be allowed by the [`PVAULT_SERVICE_ALLOWED_HTTP_DESTINATIONS`](https://docs.piiano.com/guides/configure/environment-variables#PVAULT_SERVICE_ALLOWED_HTTP_DESTINATIONS) configuration.
+   */
+  function fetch(url: string, options?: {
+    method?: string,
+    body?: string,
+    headers?: Record<string, string>
+  }): Promise<{
+    ok: boolean,
+    status: number,
+    statusText: string,
```
Comment (nirhaas) [inline sdk/vault-bundles/src/globals.ts:16]: We don't have that atm

File: sdk/vault-bundles/src/globals.ts [hunk 5]
```diff
@@ -1,8 +1,39 @@
 // Declare global functions available in Vault's Bundles.
 declare global {
-  function _httpGet(url: string): HTTPResult;
+  /**
+   * Fetches a resource from the network.
+   * The Vault fetch function implement a compatible subset of the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).
+   *
+   * API calls made with this function are subject to the same restrictions as other outbound network requests in Vault and must be allowed by the [`PVAULT_SERVICE_ALLOWED_HTTP_DESTINATIONS`](https://docs.piiano.com/guides/configure/environment-variables#PVAULT_SERVICE_ALLOWED_HTTP_DESTINATIONS) configuration.
```
Comment (nirhaas) [inline sdk/vault-bundles/src/globals.ts:7]: It's `PVAULT_SERVICE_ALLOWED_PCI_HTTP_DESTINATIONS` for now

File: examples/custom-actions/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import type {VaultFunction} from "@piiano/vault-bundles";
+
+export const send_welcome_email: VaultFunction = {
+  type: "action",
+  description: "Send welcome email to user",
+  dependencies: {
+    http_post_access: ['host.docker.internal:3000'],
```
Comment (nirhaas) [inline examples/custom-actions/src/index.ts:7]: Remove

**Review by nirhaas — APPROVED** [review]
I did not review sdk/forms, but other than that lgtm

**Pull request merged**